### DISCUSSION TITLE
EKS Module: Swap out for Karpenter

### DISCUSSION THREAD

**Discussion by iantybo** [gruntwork-io/knowledge-base#563] [Help] [answered]
Is it possible to use https://karpenter.sh/ once my Reference Architecture has been delivered with the EKS on EC2 module?

> Karpenter is an open-source node provisioning project built for Kubernetes. Karpenter improves the efficiency and cost of running workloads on Kubernetes clusters by:
Watching for pods that the Kubernetes scheduler has marked as unschedulable
Evaluating scheduling constraints (resource requests, nodeselectors, affinities, tolerations, and topology spread constraints) requested by the pods
Provisioning nodes that meet the requirements of the pods
Removing the nodes when the nodes are no longer needed

---

<ins datetime="2022-09-20T16:15:06Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109271">Tracked in ticket #109271</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
We currently do not support Karpenter as an option for the worker nodes. There is [a feature ticket in `terraform-aws-eks`]( https://github.com/gruntwork-io/terraform-aws-eks/issues/379) that you can follow for progress on implementing this.

**Comment by vramanufidato** [comment]
Is the Karpenter been supported currently ?

**Comment by andi-pangeran** [comment]
Is the Karpenter been supported currently ?
